src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -313,10 +313,24 @@ def update_existing_check_in(
                     )
                 else:
                     check_in = MonitorCheckIn.objects.select_for_update().get(
-                        monitor_environment=monitor_environment,
                         guid=check_in_id,
```

Comment:
we don't want to remove the filter from the `use_checkin_latest` logic because we need to scope the query + check-in doesn't have a provided guid regardless